Updates for server and mc releases on 2024-06-10 (#1243)
minio/docs: source/administration/object-management.rst

.. cond:: windows

>    Unlike filenames on a Windows system, object names in MinIO cannot have either a ``\`` or ``:`` character in the object name.

Reviewer: maybe a bit more obvious?
```suggestion
   Unlike filenames on a Windows system, object names in MinIO must not include either ``\`` or ``:`` characters in the object name.
```

Reviewer 2: I think we're backing into this. Suggestion:

```suggestion
MinIO does not support ``\`` or ``:`` characters in object names, regardless of support for those characters in Windows filesystems.
```

Author: Sure!

>    Unlike filenames on a Windows system, object names in MinIO must not include either ``\`` or ``:`` characters in the object name.

Reviewer 2: I think we're backing into this a bit. 

```suggestion
MinIO does not support the ``\`` or ``:`` characters in object names, regardless of support for those characters in Windows filesystems.
```
